src/aks-preview/azext_aks_preview/managed_cluster_decorator.py

```diff
@@ -2120,6 +2140,21 @@ def set_up_kube_proxy_config(self, mc: ManagedCluster) -> ManagedCluster:
         mc.network_profile.kube_proxy_config = self.context.get_kube_proxy_config()
         return mc
 
+    def set_up_auto_upgrade_profile(self, mc: ManagedCluster) -> ManagedCluster:
+        """Set up auto upgrade profile for the ManagedCluster object.
+        :return: the ManagedCluster object
+        """
+        self._ensure_mc(mc)
```

Comment:
Also you could remove this line as the same check is done in the core cli.

Reply:
Removed both instances of `self._ensure_mc(mc)`